src/SimpleList/index.js

```diff
@@ -46,9 +56,29 @@ export default class SimpleList extends Component {
     )
   }
 
+  filterElement = (query) => {
+    this.setState({ currentQuery: query })
+  }
+
+  filterItems(items) {
+    return items.filter((itm) => {
+      if (itm.firstLine.text.indexOf(this.state.currentQuery) >= 0) {
```

Comment:
I forgot to mention in your other `filterItems` functions, but try to destructure out `currentQuery` so you don't have to write `this.state` every time.